src/tests/strategies/polygon/balancer/testBalancerBase.js

```diff
@@ -0,0 +1,151 @@
+
+      await controller.withdrawAll(want.address);
+
+      let _after = await want.balanceOf(pickleJar.address);
+      console.log("Picklejar balance after controller withdrawal: ", _after.toString());
+
+      expect(_after).to.be.gt(_before, "controller withdrawAll failed");
+
+      _before = await want.balanceOf(alice.address);
+      console.log("\nAlice balance before picklejar withdrawal: ", _before.toString());
+
+      await pickleJar.cooldown((await pickleJar.balanceOf(alice.address)));
+
+      await increaseTime(60 * 60 * 24 * 7); //travel 7 days
+
+      await pickleJar.withdrawAll();
+
+      _after = await want.balanceOf(alice.address);
+      console.log("Alice balance after picklejar withdrawal: ", _after.toString());
+
+      expect(_after).to.be.gt(_before, "picklejar withdrawAll failed");
+      expect(_after).to.be.gt(_want, "no interest earned");
+    });
+
+    it("Should harvest correctly", async () => {
+      const _want = await want.balanceOf(alice.address);
+      await want.approve(pickleJar.address, _want);
+      await pickleJar.deposit(_want);
+      console.log("Alice pTokenBalance after deposit: %s\n", (await pickleJar.balanceOf(alice.address)).toString());
+      await pickleJar.earn();
+
+      const _before = await pickleJar.balance();
+      let _treasuryBefore = await want.balanceOf(treasury.address);
+
+      console.log("Picklejar balance before harvest: ", _before.toString());
+      console.log("💸 Treasury balance before harvest: ", _treasuryBefore.toString());
+      console.log("\nRatio before harvest: ", (await pickleJar.getRatio()).toString());
+
+      await rewardToken.transfer(strategy.address, toWei(1));
+
+      await strategy.harvest();
+
+      const _after = await pickleJar.balance();
+      let _treasuryAfter = await want.balanceOf(treasury.address);
+      console.log("Ratio after harvest: ", (await pickleJar.getRatio()).toString());
+      console.log("\nPicklejar balance after harvest: ", _after.toString());
+      console.log("💸 Treasury balance after harvest: ", _treasuryAfter.toString());
+
+      //20% performance fee is given
+      const earned = _after.sub(_before).mul(1000).div(800);
+      const earnedRewards = earned.mul(200).div(1000);
+      const actualRewardsEarned = _treasuryAfter.sub(_treasuryBefore);
+      console.log("\nActual reward earned by treasury: ", actualRewardsEarned.toString());
+
+      expect(earnedRewards).to.be.eqApprox(actualRewardsEarned, "20% performance fee is not given");
+
+      //withdraw
+      const _devBefore = await want.balanceOf(devfund.address);
+      _treasuryBefore = await want.balanceOf(treasury.address);
+      console.log("\n👨‍🌾 Dev balance before picklejar withdrawal: ", _devBefore.toString());
+      console.log("💸 Treasury balance before picklejar withdrawal: ", _treasuryBefore.toString());
+
+      const _cooldown = await pickleJar.balanceOf(alice.address);
+
+      await pickleJar.cooldown(_cooldown);
```

Comment:
I see that `cooldown` is set for the Alice somewhere somewhat random in the flow. Is there a particular part when it should be called? Why not bake it into the `deposit` function? See the Adamant contract, which does this.